Support EnabledControl Parameters

palaniswamy opened the pull request:
*Issue #, if available:*

*Description of changes:*
Some of the EnabledControls use parameters. Add logic to parse the parameters from the feed - constants.py


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

palaniswamy committed:
Support EnabledControl Parameters

igirardi commented on constants.py:
```diff
@@ -34,41 +34,46 @@
+            "24izmu4k16gv9tvd7sexnyrfy": {},  # AWS-GR_ROOT_ACCOUNT_MFA_ENABLED
+            "8sw3pbid15t9cbww8d2w2qwgf": {},  # AWS-GR_S3_BUCKET_PUBLIC_READ_PROHIBITED
+            "9j9nwxj789d82sypnukhyyowy": {},  # AWS-GR_S3_BUCKET_PUBLIC_WRITE_PROHIBITED
+            "ka8e3pkqefnjsxuyc26ji580": {
```
@palaniswamy thank you for your contribution with this PR. This is very useful. Please add additional 4-5 examples of parametrized controls (list of parametrized controls https://docs.aws.amazon.com/controltower/latest/controlreference/control-parameter-concepts.html) using the global identifiers from https://docs.aws.amazon.com/controltower/latest/controlreference/all-global-identifiers.html.

palaniswamy replied:
Sure @igirardi , examples are added.

igirardi commented on constants.py:
```diff
@@ -34,41 +34,46 @@
+            "1oxkwnc4hwhi2ndv6ekwy7np7": {},  # AWS-GR_REDSHIFT_CLUSTER_PUBLIC_ACCESS_CHECK
+            "6wmutsohbkwhfw6sf7cbt5e81": {},  # AWS-GR_S3_ACCOUNT_LEVEL_PUBLIC_ACCESS_BLOCKS_PERIODIC
+            "66gfl06uj1v999z53szvu0exa": {},  # AWS-GR_SAGEMAKER_NOTEBOOK_NO_DIRECT_INTERNET_ACCESS
+            "dfanrd8y5p7oj8fjyugqnakfr": {},  # AWS-GR_SSM_DOCUMENT_NOT_PUBLIC
```
@palaniswamy thank you for your contribution with this PR. This is very useful. Please add additional 4-5 examples (different than before/above) of parametrized controls (list of parametrized controls https://docs.aws.amazon.com/controltower/latest/controlreference/control-parameter-concepts.html) using the global identifiers from https://docs.aws.amazon.com/controltower/latest/controlreference/all-global-identifiers.html.

palaniswamy replied:
sure @igirardi , examples are added.

igirardi commented on stacks/aws_control_tower_guardrails_stack.py:
```diff
@@ -133,7 +133,7 @@ def enable_guardrails(self) -> None:
                 ou_id,
             ) in organizational_units_arns.items():
-                for control_name in enable_guardrails:
+                for control_name in enable_guardrails.keys():
```
@palaniswamy thank you for your contribution with this PR. This is very useful. Please make this backward compatible in the case the config file contains a set. Code should work for both old and new config files for better compatibility.

palaniswamy replied:
Sure Ivan @igirardi , backward compatibility is added.

igirardi commented on stacks/aws_control_tower_guardrails_stack.py:
```diff
@@ -142,6 +142,8 @@ def enable_guardrails(self) -> None:
                         ),
                         target_identifier=organizational_unit_arn,
                     )
+                    if enable_guardrails[control_name]:
```
@palaniswamy thank you for your contribution with this PR. This is very useful. Same here please ensure backward compatibility.

palaniswamy replied:
sure @igirardi , backward compatibility is added.

igirardi reviewed (changes requested):
Please address comments in the PR and change the format of the configuration file to the following one to have the possibility to include tags:

```
"ka8e3pkqefnjsxuyc26ji580": { # CT.MULTISERVICE.PV.1
  "Parameters": {
    "AllowedRegions": ["us-east-1", "us-west-2"],
    "ExemptedPrincipalArns": [],
    "ExemptedActions": [],
  },
  "Tags": [{"key": "Environment", "value": "Production"}]
}
```
Please update the code to both parameters and tags.

palaniswamy committed:
Backward compatability; More example controls.

palaniswamy committed:
Added support for controltower:TagResource

palaniswamy commented:
Hi @igirardi , thanks for reviewing the PR and for the comments. I added the backward compatibility, added a few examples and modified README.md a bit. You will have to update the AWS document (referenced as "Additional Information" in the README.md.). Please review again and merge if it looks good.

palaniswamy committed:
Moved the comments in constants.py as Ivan suggested.

igirardi commented:
@palaniswamy thank you for the PR. We will test it and approve early next week.

igirardi commented:
There are few fixes we will add after merging the PR to resolve permission issues and control tagging issues.
